Bug: Warning popping up during `Make data` timestamp - python related

**Describe the bug**
Making the data for development is slow and shows this error:

`/Users/vivek/github/reports/reports/generate_reports_data.py:234: FutureWarning: Comparison of Timestamp with datetime.date is deprecated in order to match the standard library behavior. In a future version these will be considered non-comparable. Use 'ts == pd.Timestamp(date)' or 'ts.date() == date' instead.                             | 144/182 [01:52<00:30,  1.27it/s]
  & (df["service_date"] >= date_start)
/Users/vivek/github/reports/reports/generate_reports_data.py:235: FutureWarning: Comparison of Timestamp with datetime.date is deprecated in order to match the standard library behavior. In a future version these will be considered non-comparable. Use 'ts == pd.Timestamp(date)' or 'ts.date() == date' instead.
  & (df["service_date"] <= date_end)`
**To Reproduce**
Steps to reproduce the behavior:
1. Go to repors/reports
2. after `poetry run make parameters`, then run `poetry run make data`

**Expected behavior**
Less errors, faster processing

**Additional context**
This must have only popped up recently.

Might be relevant. Ran the github action to deploy the dev reports site. After almost 3 hrs, job failed at `Deploy Development to Netlify... Error: Process completed with exit code 127.`

![Image](https://github.com/user-attachments/assets/215a59a1-e200-4e92-ab03-5a632749b36f)

> Might be relevant. Ran the github action to deploy the dev reports site. After almost 3 hrs, job failed at `Deploy Development to Netlify... Error: Process completed with exit code 127.`
> 
> ![Image](https://github.com/user-attachments/assets/215a59a1-e200-4e92-ab03-5a632749b36f)

This is likely due to a separate issue:
https://github.com/cal-itp/reports/issues/327

--- a/reports/generate_reports_data.py
+++ b/reports/generate_reports_data.py
@@ -139,12 +139,12 @@ def _rt_completeness():
 
 
 def generate_rt_completeness(itp_id: int, date_start: str, date_end: str):
-    df = _rt_completeness()
+    df = _rt_completeness()  # service_date format: datetime64, 'YYYY-MM-DD 00:00:00'
     date_start = pd.to_datetime(date_start)
     date_end = pd.to_datetime(date_end)
 
     # Filter the DataFrame based on the conditions
-    return df[
+    return df[  # Why doesn't this comparison fail? like the other comparisons?
         (df["calitp_itp_id"] == itp_id)
         & (df["service_date"] >= date_start)
         & (df["service_date"] <= date_end)
@@ -179,9 +179,16 @@ GROUP BY 1, 2, 3
 
 
 def generate_ave_median_tu_age(itp_id: int, date_start, date_end):
-    df = _median_tu_age()
-    date_start = pd.to_datetime(date_start)
-    date_end = pd.to_datetime(date_end)
+    """
+    Args:
+        itp_id (int): The ITP ID to filter the DataFrame.
+        date_start (str): The start date of the range in 'YYYY-MM-DD' format.
+        date_end (str): The end date of the range in 'YYYY-MM-DD' format.
+    """
+    df = _median_tu_age()  # service_date format: datetime64[ns], 2025-01-05 00:00:00
+
+    date_start = pd.Timestamp(date_start)
+    date_end = pd.Timestamp(date_end)
 
     # Filter the DataFrame based on the conditions
     return df[
@@ -224,9 +231,16 @@ ORDER BY
 
 
 def generate_ave_median_vp_age(itp_id: int, date_start, date_end):
-    df = _median_vp_age()
-    date_start = pd.to_datetime(date_start)
-    date_end = pd.to_datetime(date_end)
+    """
+    Args:
+       itp_id (int): The ITP ID to filter the DataFrame.
+       date_start (str): The start date of the range in 'YYYY-MM-DD' format.
+       date_end (str): The end date of the range in 'YYYY-MM-DD' format.
+    """
+    df = _median_vp_age()  # service_date format: object, 'YYYY-MM-DD'
+    df["service_date"] = pd.to_datetime(df["service_date"])
+    date_start = pd.Timestamp(date_start)
+    date_end = pd.Timestamp(date_end)
 
     # Filter the DataFrame based on the conditions
     return df[

